src/binary-encoding.adoc

```diff
@@ -2298,16 +2298,302 @@ considered "shapes" rather than types because while their encoding is always sta
 produced by their expansion is not. A single macro can produce streams of varying length and containing values of
 different Ion types depending on the arguments provided in the invocation.
 
-TODO: Examples
+See the link:macros-by-example.adoc#macro_shapes[Macro Shapes] section of _Macros by Example_ for more information.
 
 === Encoding E-expressions With Multiple Arguments
 
-TODO
+If a macro has more than one parameter, the E-expression arguments corresponding to each parameter are encoded one
+after the other moving from left to right.
 
-=== Grouped Parameter Encodings
+.Figure {counter:figure}: Definition of macro `_foo_` at address `_0_`
+[%unbreakable,source]
+----
+(macro foo             // Macro name
+  [                    // Parameters
+    (a string!),
+    (b compact_symbol!),
+    (c uint16!)
+  ]
+  /* ... */            // Body (elided)
+)
+----
+
+.Figure {counter:figure}: Encoding of E-expression for macro with multiple parameters: `_(:0 "hello" baz 512)_`
+[%unbreakable,source]
+----
+┌──── The opcode is less than 0x40, so it is an E-expression invoking the macro at
+│     address 0, `foo`. `foo`'s first parameter is a string, so an opcode follows.
+│
+│  ┌──── Opcode 0x85 indicates a 5-byte string. 5 UTF-8 bytes follow.
+│  │
+│  │                 ┌──── `foo`'s second parameter is a compact_symbol, so a `FlexSym` follows.
+│  │                 │     FlexSym -3: 3 bytes of UTF-8 text follow.
+│  │                 │
+│  │                 │           ┌──── `foo`'s third parameter is a uint16, so a 2-byte
+│  │                 │           │     2-byte `FixedUInt` follows.
+│  │                 │           │     FixedUInt: 512
+│  │  h  e  l  l  o  │   b  a  z │
+00 85 68 65 6C 6C 6F FD 62 61 7A 00 20
+      └──────┬─────┘    └───┬──┘
+        UTF-8 bytes    UTF-8 bytes
+----
+
+[[argument_encoding_bitmap]]
+=== Argument Encoding Bitmap (AEB)
```

Comment:
[nitpick] This no longer feels like a "bitmap", but I don't have a better term to offer ATM.

Reply:
Earlier versions of this tried "configuration," "specification" ("spec"), "header," and "prefix." I'll leave it as-is for the moment and we can change it when we think of something better.